### PR TITLE
Initial "upload from device" code

### PR DESCRIPTION
**Pull request by nixerus**

### CODE REVIEW

File: routes/utils.js [hunk 1]
```diff
@@ -57,11 +57,15 @@ function deleteFiles(files) {
...
-function downloadVideo(url, callback) {
-	let videoExtension = url.split('.').pop().toLowerCase();
-	var videoDownloadPath = path.join(__dirname, `video_${Date.now()}_${parseInt(Math.random() * 10000)}` + '.' + videoExtension);
-	var cmd = ("ffmpeg -y -i " + url + " -vcodec copy -acodec copy " + videoDownloadPath);
+function downloadVideo(url, name, callback) {
+	if ( name == null ){
+		name = url;
+	}
+	let videoExtension = name.split('.').pop().toLowerCase();
+	var videoDownloadPath = `video_${Date.now()}_${parseInt(Math.random() * 10000)}` + '.' + videoExtension
```
Comment (gopavasanth) [inline routes/utils.js]: Please change to 
path.join(__dirname, `video_${Date.now()}_${parseInt(Math.random() * 10000)}` + '.' + videoExtension);
Might get issues in production (toolforge).

**Comment by gopavasanth** [comment]
Hi @nixerus,
We deployed all our new changes of VideoCutTool [here](https://tools.wmflabs.org/video-cut-tool), I found some times Upload Viedo from the local device is failing and displaying "Something went wrong". I could find any logs in the server for those failures.

**Comment by gopavasanth** [comment]
Hi @nixerus,

Most of the uploaded videos from the device will be in mp4 format which commons will not accept while uploading, So I would recommend converting those uploaded videos to Wikimedia commons acceptable formates.

[T244909](https://phabricator.wikimedia.org/T244909)

**Comment by nixerus** [comment]
Hi @gopavasanth , sorry I only just saw your original comment.

Do you think that the error mentioned prior is being caused by the uploaded videos being in an MP4 format or is that an unrelated issue?

**Comment by gopavasanth** [comment]
I don't think because some videos are not uploading to VideoCutTool in production, so it is displaying this error I guess. Did you try in production? If not I would recommend you to give a try testing in production. https://tools.wmflabs.org/video-cut-tool/

**Comment by gopavasanth** [comment]
Hey @nixerus, Any updates?

**Pull request merged**